frame/staking/src/lib.rs

```diff
@@ -2654,10 +2654,20 @@ impl<T: Trait> Module<T> {
 		let desired_winners = Self::validator_count().min(snapshot_validators_length);
 		ensure!(winners.len() as u32 == desired_winners, Error::<T>::PhragmenBogusWinnerCount);
 
+		let snapshot_nominators_len = <SnapshotNominators<T>>::decode_len()
+			.map(|l| l as u32)
+			.ok_or_else(|| Error::<T>::SnapshotUnavailable)?;
+
 		// decode snapshot validators.
 		let snapshot_validators = Self::snapshot_validators()
 			.ok_or(Error::<T>::SnapshotUnavailable)?;
 
+		// rest of the size of the solution must be correct.
+		ensure!(
+			snapshot_nominators_len == election_size.nominators,
+			Error::<T>::PhragmenBogusElectionSize,
+		);
+
```

Comment:
```suggestion
		let snapshot_nominators_len = <SnapshotNominators<T>>::decode_len()
			.map(|l| l as u32)
			.ok_or_else(|| Error::<T>::SnapshotUnavailable)?;

		// rest of the size of the solution must be correct.
		ensure!(
			snapshot_nominators_len == election_size.nominators,
			Error::<T>::PhragmenBogusElectionSize,
		);

		// decode snapshot validators.
		let snapshot_validators = Self::snapshot_validators()
			.ok_or(Error::<T>::SnapshotUnavailable)?;

```

I would put the length check even earlier before doing any decoding.

Reply:
damn this diff looks terrible, but I just meant moving the nominator `decode_len` directly after the validator `decode_len`